### PR TITLE
Added a keep feature to projection transform plugin CDAP-6067 (Doc Update)

### PR DESCRIPTION
**Pull request by riwazp7**
Updated Projection-transform.md to reflect the added keep feature.

### CODE REVIEW

File: core-plugins/docs/Projection-transform.md [hunk 1]
```diff
@@ -3,8 +3,8 @@
 
 Description
 -----------
-The Projection transform lets you drop, rename, and cast fields to a different type.
-Fields are first dropped, then cast, then renamed.
+The Projection transform lets you drop, keep, rename, and cast fields to a different type.
+Fields are first dropped based on drop or keep field, then cast, then renamed.
```
Comment (awholegunch) [inline core-plugins/docs/Projection-transform.md]: based on the drop or keep field,

File: core-plugins/docs/Projection-transform.md [hunk 2]
```diff
@@ -32,21 +32,27 @@ and then field 'A' will be renamed to 'B':
     | B          | int  | 10     |
     +============================+
 
+Similarly, the transfrom will first check if it should keep a field, then rename it if configured to do so.
```
Comment (awholegunch) [inline core-plugins/docs/Projection-transform.md]: should keep a field, and then rename it if configured to do so.

File: core-plugins/docs/Projection-transform.md [hunk 3]
```diff
@@ -32,21 +32,27 @@ and then field 'A' will be renamed to 'B':
...
 
 Use Case
 --------
-The transform is used when you need to drop fields, change field types, or rename fields.
+The transform is used when you need to drop fields, keep specific fields, change field types, or rename fields.
 
 For example, you may want to rename a field from ``'timestamp'`` to ``'ts'`` because you want
-to write to a database where ``'timestamp'`` is a reserved keyword. Or, you might want to
+to write to a database where ``'timestamp'`` is a reserved keyword, or you might want to
```
Comment (awholegunch) [inline core-plugins/docs/Projection-transform.md]: is a reserved keyword. You might want to

File: core-plugins/docs/Projection-transform.md [hunk 4]
```diff
@@ -32,21 +32,27 @@ and then field 'A' will be renamed to 'B':
...
 drop a field named ``'headers'`` because you know it is always empty for your particular
-data source. 
+data source. Or, you might want only keep fields named ``'ip'`` and ``'timestamp'`` and discard
```
Comment (awholegunch) [inline core-plugins/docs/Projection-transform.md]: Or, you might want to only keep fields named

File: core-plugins/docs/Projection-transform.md [hunk 5]
```diff
@@ -32,21 +32,27 @@ and then field 'A' will be renamed to 'B':
...
-data source. 
+data source. Or, you might want only keep fields named ``'ip'`` and ``'timestamp'`` and discard 
+all other fields.
 
 
 Properties
 ----------
 **drop:** Comma-separated list of fields to drop. For example: ``'field1,field2,field3'``.
 
+**keep:** Comma-seperated list of fields to keep. For example: ``'field1,field2,field3'``.
```
Comment (awholegunch) [inline core-plugins/docs/Projection-transform.md]: Comma-separated

File: core-plugins/docs/Projection-transform.md [hunk 6]
```diff
@@ -32,21 +32,27 @@ and then field 'A' will be renamed to 'B':
...
+
+Note: Both drop and keep fields can *not* be specified. At least one must be null or empty.
```
Comment (awholegunch) [inline core-plugins/docs/Projection-transform.md]:
```
Note: Drop and keep fields cannot *both* be specified. At least one must be either not defined, null, or an empty string.
```

File: core-plugins/docs/Projection-transform.md [hunk 7]
```diff
@@ -68,16 +74,17 @@ converted to an int.
 
 Example
 -------
-This example drops the ``'ts'`` and ``'headers'`` fields. It also changes the type of the ``'cost'``
-field to a double and renames it ``'price'``. 
+This example only keeps the ``'id'`` and ``'cost'`` fields. It also changes the type of the ``'cost'``
```
Comment (awholegunch) [inline core-plugins/docs/Projection-transform.md]: This example keeps only the

File: core-plugins/docs/Projection-transform.md [hunk 8]
```diff
@@ -68,16 +74,17 @@ converted to an int.
...
     {
         "name": "Projection",
         "type": "transform",
         "properties": {
-            "drop": "ts,headers",
+            "drop": "",
             "convert": "cost:double",
-            "rename": "cost:price"
+            "rename": "cost:price",
+	    "keep": "id,cost"
```
Comment (awholegunch) [inline core-plugins/docs/Projection-transform.md]: indent one space to match preceding line

File: core-plugins/src/main/java/co/cask/hydrator/plugin/transform/ProjectionTransform.java [hunk 9]
```diff
@@ -151,10 +161,20 @@ public void transform(StructuredRecord valueIn, Emitter<StructuredRecord> emitte
   }
 
   private void init() {
+
+    if (!Strings.isNullOrEmpty(projectionTransformConfig.drop) &&
+    !Strings.isNullOrEmpty(projectionTransformConfig.keep)) {
+      throw new IllegalArgumentException("Cannot specify both drop and keep. One should be empty or null");
```
Comment (awholegunch) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/transform/ProjectionTransform.java]: Add trailing period:

```
One should be empty or null.
```

**Comment by awholegunch** [comment]
I can take another look after these changes have been made.

**Comment by riwazp7** [comment]
Done. Can you take another look?

**Comment by awholegunch** [comment]
One comment (https://github.com/caskdata/hydrator-plugins/pull/275/files#r66902491) to be addressed, otherwise once that is done, it LGTM.

File: core-plugins/docs/Projection-transform.md (hunk 8 above)
Comment (riwazp7) [inline core-plugins/docs/Projection-transform.md]: Hmm the indent looks fine when I open it to edit but looks out of place when viewed.

**Comment by riwazp7** [comment]
This is an LGTM according to jj. Waiting for the code to get merged before I merge the docs.

**Pull request merged**